prereise/gather/griddata/hifld/data_access/load.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+import tempfile
+from io import BytesIO
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import pandas as pd
+
+from prereise.gather.griddata.hifld.const import abv2state  # noqa F401
+
+
+def get_hifld_power_plants(path):
+    """Read the HIFLD Power Plants CSV file and keep operational plants located in
+    contiguous states.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- operational power plant in contiguous states.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'OP' and STATE in @abv2state")
+
+
+def get_hifld_generating_units(path):
+    """Read the HIFLD Generating Units CSV file and keep operational plants located in
+    contiguous states.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- operational generating units in contiguous states.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'OP' and STATE in @abv2state")
+
+
+def get_hifld_electric_substations(path):
+    """Read the HIFLD Electric Substations CSV file and keep in service substations
+    located in contiguous states and connected to at least one transmission line.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- in service electric substations in contiguous
+        states that are connected to at least one electric power transmission line.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'IN SERVICE' and STATE in @abv2state and LINES != 0")
+
+
+def get_hifld_electric_power_transmission_lines(path):
+    """Read the HIFLD Electric Power Transmission Lines CSV file and keep in service
+    lines.
+
+    :param str path: path to file. Either local or URL.
+    :return: (*pandas.DataFrame*) -- in service electric power transmission lines.
+    """
+    data = pd.read_csv(path)
+    return data.query("STATUS == 'IN SERVICE'")
+
+
+def get_hifld_electric_power_transmission_lines_topology(path):
+    """Read the HIFLD Electric Power Transmission Lines json zip file and keep in
+    service lines.
+
+    :param str path: path to zip file. Either local or URL.
+    :return: (*list*) -- each element is a dictionary enclosing the information of the
```

Comment:
Should we return this one as a dataframe, to match all the others?

Reply:
not a bad idea

Reply:
Done. The coordinates of the lines were missing for some reason. It should be good now:
```
>>> from prereise.gather.griddata.hifld.data_access.load import get_hifld_electric_power_transmission_lines_topology
>>> topology = get_hifld_electric_power_transmission_lines_topology("https://besciences.blob.core.windows.net/datasets/hifld/Electric_Power_Transmission_Lines_Jul2020.geojson.zip")
>>> topology
       OBJECTID      ID          TYPE      STATUS  ...                  SUB_1          SUB_2 SHAPE_Length                                        coordinates
0             1  212144  AC; OVERHEAD  IN SERVICE  ...         EAST ARLINGTON      TAP209798     0.090149  [[-122.11300327699996, 48.18834609800007], [-1...
1             2  212145  AC; OVERHEAD  IN SERVICE  ...              SNOHOMISH  UNKNOWN201930     0.002900  [[-122.10387420599994, 47.923777407000046], [-...
2             3  212146  AC; OVERHEAD  IN SERVICE  ...                  MILAN      TAP209809     0.065357  [[-117.3523942, 47.83782039300007], [-117.3521...
3             4  212147  AC; OVERHEAD  IN SERVICE  ...                   TOLT      TAP209799     0.000507  [[-121.90700143999999, 47.647450237000044], [-...
4             5  212148  AC; OVERHEAD  IN SERVICE  ...  OLYMPIC VAIL PIPELINE      TAP209800     0.011773  [[-122.69515214199998, 46.87299534200008], [-1...
...         ...     ...           ...         ...  ...                    ...            ...          ...                                                ...
89739     89740  175393      OVERHEAD  IN SERVICE  ...            RISER173638  UNKNOWN132385     0.025167  [[-74.20126511499996, 39.66126657700005], [-74...
89740     89741  175394      OVERHEAD  IN SERVICE  ...               MANITOUS    RISER173640     0.119295  [[-74.22001359299998, 39.952481841000065], [-7...
89741     89742  175395      OVERHEAD  IN SERVICE  ...            RISER173641  UNKNOWN172406     0.020439  [[-74.07734, 39.943226], [-74.07755150899999, ...
89742     89743  175396      OVERHEAD  IN SERVICE  ...            RISER173640    RISER173641     0.017860  [[-74.09688414199996, 39.944998197000075], [-7...
89743     89744  175397      OVERHEAD  IN SERVICE  ...              OCEANVIEW  UNKNOWN173642     0.021345  [[-74.02747412899998, 40.222604352000076], [-7...

[71554 rows x 18 columns]
```